src/bbs2gh/AwsApiFactory.cs

```diff
@@ -0,0 +1,19 @@
+namespace OctoshiftCLI.BbsToGithub;
+
+public class AwsApiFactory
+{
+    private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+    public AwsApiFactory(EnvironmentVariableProvider environmentVariableProvider)
+    {
+        _environmentVariableProvider = environmentVariableProvider;
+    }
+
+    public virtual AwsApi Create(string awsAccessKey = null, string awsSecretKey = null)
+    {
+        var accessKey = _environmentVariableProvider.AwsAccessKey() ?? awsAccessKey;
+        var secretKey = _environmentVariableProvider.AwsSecretKey() ?? awsSecretKey;
```

Comment:
These won't work as intended and should be inverted because if the env variable is not set it is going to throw:
```suggestion
        var accessKey = awsAccessKey ?? _environmentVariableProvider.AwsAccessKey();
        var secretKey = awsSecretKey ?? _environmentVariableProvider.AwsSecretKey();
```